src/olympia/addons/models.py

```diff
@@ -2385,3 +2386,24 @@ class Meta:
     def save(self, *args, **kwargs):
         self.hashed_guid = hashlib.sha256(self.guid.encode()).hexdigest()
         super().save(*args, **kwargs)
+
+
+class AddonBrowserMapping(ModelBase):
+    """
+    Mapping between a Firefox add-on on AMO and an extension ID for a different
+    browser.
+    """
+
+    addon = models.ForeignKey(
+        Addon,
+        on_delete=models.CASCADE,
+        help_text='Add-on id this item will point to. If you do not know the '
+        'id, paste the slug instead and it will be transformed automatically '
+        'for you. You can also use the magnifying glass to see all the '
+        'available add-ons if you have access to the add-on admin page.',
+    )
+    browser = models.PositiveSmallIntegerField(choices=BROWSERS.items())
+    extension_id = models.CharField(max_length=255, null=False, db_index=True)
+
+    class Meta:
+        unique_together = ('addon', 'browser', 'extension_id')
```

Comment:
I think the actual unique constraint we want is `browser`, `extension_id`. We shouldn't need to include the `addon` in there, we never want the same browser + extension id combo to point to multiple add-ons.

(See also nitpicking above for the docstring)